cmd/api/api.go

```diff
@@ -136,3 +151,114 @@ func checkDB(database *database.SQLdb, timeout time.Duration) error {
 
 	return database.DB.PingContext(ctx)
 }
+
+// getFiles returns the files from the database for a specific user
+func getFiles(c *gin.Context) {
+
+	log.Debugf("request files in project")
+	c.Writer.Header().Set("Content-Type", "application/json")
+	// Get user ID to extract all files
+	userID, err := getUserFromToken(c.Writer, c.Request)
+	if err != nil {
+		// something went wrong with user token
+		c.JSON(500, err.Error())
+
+		return
+	}
+
+	files, err := Conf.API.DB.GetUserFiles(userID)
+	if err != nil {
+		// something went wrong with querying or parsing rows
+		c.JSON(500, err.Error())
+
+		return
+	}
+
+	// Return response
+	c.JSON(200, files)
+}
+
+// getUserFromToken parses the token, validates it against the key and returns the key
```

Comment:
i think the token check should be split, so that it is easier to integrate with sda download: see: https://github.com/neicnordic/sda-download/blob/main/pkg/auth/auth.go#L54-L90

Also it would be good to have a list of trusted issuers: https://github.com/neicnordic/sda-download/blob/main/pkg/auth/auth.go#L327-L344